packages/ag-charts-community/src/chart/series/cartesian/scatterSeries.ts

```diff
@@ -149,11 +160,15 @@ export class ScatterSeries extends CartesianSeries<SeriesNodeDataContext<Scatter
 
         const { dataModel, processedData } = await dataController.request<any, any, true>(this.id, data ?? [], {
             props: [
+                keyProperty(this, xKey, isContinuousX, { id: 'xKey-raw' }),
```

Comment:
Aren't these just value properties?

Reply:
They are, but they are needed for the diff processor to assign a unique id to each datum. So it becomes a composite key.

See here https://github.com/ag-grid/ag-charts/blob/latest/packages/ag-charts-community/src/chart/data/processors.ts#L269-L270

Admittedly, there may be a better way.